src/transformers/utils/import_utils.py

```diff
@@ -39,16 +39,32 @@
 
 # TODO: This doesn't work for all packages (`bs4`, `faiss`, etc.) Talk to Sylvain to see how to do with it better.
 def _is_package_available(pkg_name: str, return_version: bool = False) -> Union[Tuple[bool, str], bool]:
-    # Check we're not importing a "pkg_name" directory somewhere but the actual library by trying to grab the version
+    # Check if the package spec exists and grab its version to avoid importing a local directory
     package_exists = importlib.util.find_spec(pkg_name) is not None
     package_version = "N/A"
     if package_exists:
         try:
+            # Primary method to get the package version
             package_version = importlib.metadata.version(pkg_name)
-            package_exists = True
```

Comment:
Why remove this line? Don't we still want this to be set to true if we correctly parse the version?

Reply:
this is because we are under `if package_exists:` so at that point, it must be true already. Setting it doesn't do anything thing (logically)

Reply:
Ah, yeah, you're right